csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/shifter/SweCsaNetworkShifter.java

```diff
@@ -0,0 +1,182 @@
+                BUSINESS_LOGS.info("Applying shift iteration {} ", iterationCounter);
+                for (Map.Entry<String, Double> entry : scalingValuesByCountry.entrySet()) {
+                    String zoneId = entry.getKey();
+                    double asked = entry.getValue();
+                    String logApplyingVariationOnZone = String.format("Applying variation on zone %s (target: %.2f)", zoneId, asked);
+                    BUSINESS_LOGS.info(logApplyingVariationOnZone);
+                    ScalingParameters scalingParameters = new ScalingParameters();
+                    scalingParameters.setPriority(ScalingParameters.Priority.RESPECT_OF_VOLUME_ASKED);
+                    scalingParameters.setReconnect(true);
+                    double done = zonalScalable.getData(zoneId).scale(network, asked, scalingParameters);
+                    if (Math.abs(done - asked) > shiftEpsilon) {
+                        String logWarnIncompleteVariation = String.format("Incomplete variation on zone %s (target: %.2f, done: %.2f)", zoneId, asked, done);
+                        BUSINESS_WARNS.warn(logWarnIncompleteVariation);
+                        limitingCountries.add(zoneId);
+                    }
+                }
+                if (!limitingCountries.isEmpty()) {
+                    StringJoiner sj = new StringJoiner(", ", "There are Glsk limitation(s) in ", ".");
+                    limitingCountries.forEach(sj::add);
+                    BUSINESS_WARNS.error("{}", sj);
+                    throw new GlskLimitationException(sj.toString());
+                }
+
+                // Step 2: Compute exchanges mismatch
+                LoadFlowResult result = LoadFlow.run(network, workingVariantCopyId, LocalComputationManager.getDefault(), LoadFlowParameters.load());
+                if (!result.isOk()) {
+                    BUSINESS_WARNS.error("Loadflow computation diverged on network '{}'", network.getId());
+                    BUSINESS_WARNS.error("Loadflow computation diverged on network during balancing adjustment");
+                    throw new ShiftingException("Loadflow computation diverged during balancing adjustment");
+                }
+                bordersExchanges = CountryBalanceComputation.computeSweBordersExchanges(network);
+                double mismatchEsPt = targetExchanges.get(ES_PT) - bordersExchanges.get(ES_PT);
+                double mismatchEsFr = targetExchanges.get(ES_FR) - bordersExchanges.get(ES_FR);
+
+                // Step 3: Checks balance adjustment results
+                if (Math.abs(mismatchEsPt) < DEFAULT_TOLERANCE_ES_PT && Math.abs(mismatchEsFr) < DEFAULT_TOLERANCE_ES_FR) {
+                    String logShiftSucceded = String.format("Shift succeed after %s iteration ", ++iterationCounter);
+                    BUSINESS_LOGS.info(logShiftSucceded);
+                    BUSINESS_LOGS.info("Shift succeed after {} iteration ", ++iterationCounter);
```

Comment:
Above, with the "zone (done - asked) > shiftEpsilon", the variation for each zone is compute separately, but here, we perform a loadflow and check the result on each border after the loadflow, which could be different because the loadflow is global, not limited to one zone.

Reply:
i mean
BUSINESS_LOGS.info(logShiftSucceded);
BUSINESS_LOGS.info("Shift succeed after {} iteration ", ++iterationCounter);
does the same thing